[Fix] translate with strict falsy

NicolasRoehm opened the pull request.

NicolasRoehm commented:
Hi Vincent, can you review this issue please ? I really need that feature asap :)

RinkAttendant6 commented:
All of the tests are currently failing, and this pull request does not explain what it is fixing since the previous tests were passing.

NicolasRoehm commented:
I changed the location of the `if (!$this->settings->getStrict())` statement because I failed the first implementation. I also added this condition to the `_f` function. Finally, I fixed the strict setting test.
